src/app/components/analytics/host-analytics/host-analytics.component.ts

```diff
@@ -0,0 +1,186 @@
+        (response) => {
+          this.globalService.stopLoader();
+          this.totalChallengeTeams = response.participant_team_count;
+        },
+        (err) => {
+          this.errCallBack(err);
+        },
+        () => {}
+      );
+
+      this.globalService.startLoader(`Fetching phases for challenge ${this.challengeId}`);
+      this.apiService.getUrl('challenges/challenge/' + this.challengeId + '/challenge_phase').subscribe(
+        (response) => {
+          this.globalService.stopLoader();
+          this.currentPhase = response.results;
+          const challengePhaseId = [];
+
+          for (let phaseCount = 0; phaseCount < this.currentPhase.length; phaseCount++) {
+            challengePhaseId.push(this.currentPhase[phaseCount].id);
+
+            // tslint:disable-next-line:max-line-length
+            this.apiService.getUrl('analytics/challenge/' + this.challengeId + '/challenge_phase/' + this.currentPhase[phaseCount].id + '/analytics')
+              .subscribe(
+                (res) => {
+                  for (let i = 0; i < challengePhaseId.length; i++) {
+                    if (challengePhaseId[i] === res.challenge_phase) {
+                      this.totalSubmission[challengePhaseId[i]] = res.total_submissions;
+                      this.totalParticipatedTeams[challengePhaseId[i]] = res.participant_team_count;
+                      break;
+                    }
+                  }
+                },
+                (err) => {
+                  this.errCallBack(err);
+                },
+                () => {}
+              );
+          }
+
+          for (let phaseCount = 0; phaseCount < this.currentPhase.length; phaseCount++) {
+            // challengePhaseId.push(this.currentPhase[phaseCount].id);
+            // tslint:disable-next-line:max-line-length
+            this.apiService.getUrl('analytics/challenge/' + this.challengeId + '/challenge_phase/' + this.currentPhase[phaseCount].id + '/last_submission_datetime_analysis/')
+              .subscribe(
+                (res) => {
+                  for (let i = 0; i < challengePhaseId.length; i++) {
+                    if (challengePhaseId[i] === res.challenge_phase) {
+                      this.lastSubmissionTime[challengePhaseId[i]] = res.last_submission_timestamp_in_challenge_phase;
+                      break;
+                    }
+                  }
+                },
+                (err) => {
+                  this.errCallBack(err);
+                },
+                () => {}
+              );
+          }
+
+        },
+        (err) => {
+          this.errCallBack(err);
+        },
+        () => {}
+      );
+
+
+      for (let i = 0; i < this.challengeList.length; i++) {
+        // tslint:disable-next-line:triple-equals
+        if (this.challengeList[i]['id'] == this.challengeId) {
+          this.currentChallengeDetails = this.challengeList[i];
```

Comment:
In **JS**, we should always use **===** to check for equality. If a type conversion is needed, do it before this.

Reply:
that has been done